### PR TITLE
Add mimetype and redirect url to HTTP info dict

### PR DESCRIPTION
**Pull request by joouha**
This will allow discovery of a file's mime-type and redirect URL without unnecessary additional HTTP requests.

### CODE REVIEW

File: fsspec/implementations/http.py [hunk 1]
```diff
@@ -847,6 +847,11 @@ async def _file_info(url, session, size_policy="head", **kwargs):
         elif "Content-Range" in r.headers:
             info["size"] = int(r.headers["Content-Range"].split("/")[1])
 
+        if "Content-Type" in r.headers:
+            info["mime"] = r.headers["Content-Type"].partition(";")[0]
```
Comment (martindurant) [inline fsspec/implementations/http.py:851]: It is possible this header doesn't exist

**Review by martindurant — COMMENTED** [review]
Please add a test showing this being passed through.

File: fsspec/implementations/http.py (hunk 1 above)
Comment (joouha) [reply fsspec/implementations/http.py:851]: This is presumably taken care of by the `if` statement.
Would you prefer `mime` was set to `None` if the header was missing?

Comment (martindurant) [reply fsspec/implementations/http.py:851]: Ah indeed. First day back...
By the way, what does the ";" do, how do you expect this to format?

Comment (joouha) [reply fsspec/implementations/http.py:851]: The [`Content-Type`](https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/Content-Type) HTTP header can take the following form:

```
Content-Type: text/html; charset=utf-8
Content-Type: multipart/form-data; boundary=something
```

so partitioning on the first `;` selects the mimetype from the value

Comment (martindurant) [reply fsspec/implementations/http.py:851]: OK. Is there any need for verification on that? I suppose we can just pass on the string and any client might do that.

Comment (joouha) [reply fsspec/implementations/http.py]: I agree it makes sense to just pass on the string and leave verification to the client

**Comment by joouha** [comment]
I've added tests

**Pull request merged**